### ISSUE TITLE
eva-large模型在evaluate时出现loss为nan

### ISSUE THREAD

**Issue by BaiMeiyingxue** [Mrwangkedong/EVA2.0_fintune#1]
您好，我在对large模型执行eva_inference_static.sh时，参数如下

python -m torch.distributed.launch --master_port 51234 --nproc_per_node 1 /home/zhangwenjuan/zhanghao/EVA-huggingfac   e//src/eva_evaluate.py --eval-batch-size 8 --enc-seq-length 128 --dec-seq-length 128 --save /home/zhangwenjuan/zhang   hao/EVA-huggingface//large-results/inference_static/ --log-file /home/zhangwenjuan/zhanghao/EVA-huggingface//large-r   esults/inference_static//log.txt --load /home/zhangwenjuan/zhanghao/EVA-huggingface//checkpoints/eva2.0-large --data   -path /home/zhangwenjuan/zhanghao/EVA-huggingface//data/kdconv --distributed-backend nccl --eval-generation --num-be   ams 1 --temperature 0.9 --top_k 0 --top_p 0.9 --length-penalty 1.6 --repetition-penalty 1.6
using world size: 1 and model-parallel size: 1
arguments:
  model_config ................. None
  model_parallel_size .......... 1
  fp16 ......................... False
  do_train ..................... False
  do_valid ..................... False
  do_eval ...................... False
  train_ratio .................. 1
  valid_ratio .................. 1
  test_ratio ................... 1
  batch_size ................... 4
  gradient_accumulation_steps .. 1
  train_iters .................. -1
  epochs ....................... 3
  weight_decay ................. 0.01
  checkpoint_activations ....... False
  checkpoint_num_layers ........ 1
  clip_grad .................... 1.0
  seed ......................... 422
  lr_decay_style ............... linear
  lr ........................... 0.0001
  warmup ....................... 0.01
  load ......................... /home/zhangwenjuan/zhanghao/EVA-huggingface//checkpoints/eva2.0-large
  load_optimizer_states ........ False
  load_lr_scheduler_states ..... False
  no_load_strict ............... False
  save ......................... /home/zhangwenjuan/zhanghao/EVA-huggingface//large-results/inference_static/
  save_interval ................ 10
  log_file ..................... /home/zhangwenjuan/zhanghao/EVA-huggingface//large-results/inference_static//log.txt
  log_interval ................. 5
  distributed_backend .......... nccl
  local_rank ................... 0
  eval_batch_size .............. 8
  eval_interval ................ 10
  eval_generation .............. True
  do_sample .................... False
  temperature .................. 0.9
  top_p ........................ 0.9
  top_k ........................ 0
  max_generation_length ........ 128
  min_generation_length ........ 2
  num_beams .................... 1
  no_repeat_ngram_size ......... 3
  repetition_penalty ........... 1.6
  early_stopping ............... False
  length_penalty ............... 1.6
  rule_path .................... None
  data_path .................... /home/zhangwenjuan/zhanghao/EVA-huggingface//data/kdconv
  cache_path ................... None
  tokenizer_path ............... None
  data_ext ..................... .txt
  num_workers .................. 2
  enc_seq_length ............... 128
  dec_seq_length ............... 128
  cuda ......................... True
  rank ......................... 0
  world_size ................... 1

...
****** context: 外文名字有Our Little Sister,Notre petite sur等,这是哪个公司发行的呢?发行公司是GAGA Communications,东宝国出品时间是2015年,影片获得过哪些奖项呢?只获得过第68届戛纳电影节金棕榈奖提名,知道还有哪些其它译名吗?比如海街女孩日记,这是
****** response: 出品时间是2015年,出品公司是哪家呢?
****** generation: <s></s>

Eval result:  dist-3: 3.8407 |  dist-4: 5.0181 |  bleu-4: 0.01546 |  f1: 0.60755 |  rouge-l: 0.47293 |  rouge-1: 13.359 |  rouge-2: 2.4037 |  loss: nan |

但是使用eva_base模型，什么都没改动过，就会表现得很正常，不会出现loss nan情况，可以请教下您吗？

**Comment by Mrwangkedong** [comment]
你好，我当初也出现这个问题，当改变beam_num的时候会消失，正常生成。这个github project程序是有bug的，并不完善，敬请谅解。finetune后的结果与原来非hugginface的模型差距很大，请使用时自行斟酌。

**Comment by BaiMeiyingxue** [comment]
> 你好，我当初也出现这个问题，当改变beam_num的时候会消失，正常生成。这个github project程序是有bug的，并不完善，敬请谅解。finetune后的结果与原来非hugginface的模型差距很大，请使用时自行斟酌。
> 
> 这是我在模型仓库下的issue [thu-coai/EVA#70](https://github.com/thu-coai/EVA/issues/70)

对，我是从这个问题下面找到您的开源的，非常感谢您的工作！
对于zero-shot与fineune这我之前有查资料但是很模糊，但是由于条件的有限并没有自己运行下了解更深刻些，我想请教下，在这个项目中在kdconv数据集运行eva_evaluate.py得到的指标属于zero-shot，而kdconv数据集经过eva_finetune.py运行后得到的指标属于finetune，这样理解对嘛？

**Comment by Mrwangkedong** [comment]
对的



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2023年1月18日(周三) 上午8:28
收件人: ***@***.***&gt;;
抄送: "Tony ***@***.******@***.***&gt;;
主题: Re: [Mrwangkedong/EVA2.0_fintune] eva-large模型在evaluate时出现loss为nan (Issue #1)




  
你好，我当初也出现这个问题，当改变beam_num的时候会消失，正常生成。这个github project程序是有bug的，并不完善，敬请谅解。finetune后的结果与原来非hugginface的模型差距很大，请使用时自行斟酌。
 
这是我在模型仓库下的issue thu-coai/EVA#70
  
对，我是从这个问题下面找到您的开源的，非常感谢您的工作！
 对于zero-shot与fineune这我之前有查资料但是很模糊，但是由于条件的有限并没有自己运行下了解更深刻些，我想请教下，在这个项目中在kdconv数据集运行eva_evaluate.py得到的指标属于zero-shot，而kdconv数据集经过eva_finetune.py运行后得到的指标属于finetune，这样理解对嘛？

**Comment by BaiMeiyingxue** [comment]
> 对的
> […](#)
> ---原始邮件--- 发件人: ***@***.***&gt; 发送时间: 2023年1月18日(周三) 上午8:28 收件人: ***@***.***&gt;; 抄送: "Tony ***@***.******@***.***&gt;; 主题: Re: [Mrwangkedong/EVA2.0_fintune] eva-large模型在evaluate时出现loss为nan (Issue #1) 你好，我当初也出现这个问题，当改变beam_num的时候会消失，正常生成。这个github project程序是有bug的，并不完善，敬请谅解。finetune后的结果与原来非hugginface的模型差距很大，请使用时自行斟酌。 这是我在模型仓库下的issue [thu-coai/EVA#70](https://github.com/thu-coai/EVA/issues/70) 对，我是从这个问题下面找到您的开源的，非常感谢您的工作！ 对于zero-shot与fineune这我之前有查资料但是很模糊，但是由于条件的有限并没有自己运行下了解更深刻些，我想请教下，在这个项目中在kdconv数据集运行eva_evaluate.py得到的指标属于zero-shot，而kdconv数据集经过eva_finetune.py运行后得到的指标属于finetune，这样理解对嘛？ — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

好的，谢谢啦
